### ISSUE TITLE
Listen for new messages in all conversations

### ISSUE THREAD

**Issue by saulmc** [xmtp/xmtp-js#92]
## Problem

Usually, developers need to listen for new messages in all of a user's conversations. The client is currently only capable of streaming messages from an individual conversation. Therefore, developers must aggregate conversation streams in their applications. To avoid duplicate effort, this capability should exist in the client.

### Requirements
- Stream incoming messages for all conversations that have started
- Stream incoming messages for new conversations that started since the stream was initiated

### Non-goals
- Provide control over whether inbound or outbound messages are returned

**Comment by elisealix22** [comment]
Wanted to provide a status update on this! Last week @neekolas and I fixed a couple of issues in `go-waku` around unsubscribing to `Stream`s that were blocking the second requirement of streaming incoming messages for new conversations:
- https://github.com/xmtp/go-waku/pull/29
- https://github.com/xmtp/go-waku/pull/32

Now that those are unblocked, I restarted feature development today and went down the path of establishing two streams: one to listen for existing conversations' messages and one to listen for new conversations' messages. The idea was that we could detect a new conversation in one stream, unsubscribe the previous list of conversation topics in the other stream and then resubscribe appending the newly detected conversation address from the first stream. However, I'm fearing that this is actually quite complex because of having to merge the output of two async iterators.

Instead of pushing that further, I'm going back to the drawing board to try a new approach where we only have a single stream to listen for all new and existing conversations. The hope is that we can differentiate between the two when new messages come in so we can correctly unsubscribe / resubscribe in order to append a new dm topic when a new conversation is detected (to prevent missing more messages for that convo once it's been started).

I'll report back again tomorrow on if this a feasible approach!
